### PR TITLE
[#29] [#30] [#32] [UI] [UI] As a user, I can see Multiple Choice, Emoticon, Text Field Form Question

### LINKED ISSUE #29: [UI] As a user, I can see Emoticon Selection Survey Question
## Why

* User needs to see survey question and emoticon selection 

## Acceptance Criteria

* Show question with order/total number
* Show emoticon selection
* Selected emoticon must have opacity 1.0, Unselected emoticons must have opacity 0.5
* Show the next question button
* Show submit button if the last question

## Design

Screen Shot 2022-05-10 at 09 11 33

### LINKED ISSUE #30: [UI] As a user, I can see Multiple Choice Survey Question
## Why

* User need to see survey question and multiple choice 

## Acceptance Criteria

* Show question with order/total number
* Show multiple choice
* Show next question button
* Show submit button if last question

## Design

Screen Shot 2022-05-10 at 09 01 20

### LINKED ISSUE #32: [UI] As a user, I can see Contact Form Survey Question
## Why

* User need to see survey question and contact information form

## Acceptance Criteria

* Show question with order/total number
* Show contact information form includes: Name, Email, Mobile Number
* Show next question button
* Show submit button if last question

## Design

Screen Shot 2022-05-10 at 09 01 23

### PR DESCRIPTION
**Pull request by ducbm051291**
- Close #29 
- Close #30 
- Close #32 

## What happened 👀

Create EmoticonView, MultipleChoiceView, MultipleChoiceOptionUIModel, TextFieldFormView
Add some UI constants in Dimensions
Integrate UI with SurveyQuestionScreenBuilder

## Insight 📝

- N/A

## Proof Of Work 📹

CleanShot 2023-05-10 at 16 36 50

### CODE REVIEW

File: lib/ui/survey_question/survey_answer_views/emoticon_view.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+class _EmoticonViewState extends State<EmoticonView> {
+  int? selectedIndex;
+
+  List<Widget> get _likertButtons {
+    List<Widget> widgets = [];
+    widget.type.icons.asMap().forEach((index, icon) {
+      widgets.add(
+        GestureDetector(
```
Comment (nmint8m) [inline lib/ui/survey_question/survey_answer_views/emoticon_view.dart:27]: @ducbm051291 @thiennguyen0196 

Optional: There are multiple way to adopt tapping on a widget. By using `GestureDetector` widget or `InkWell` widget. You guys can try it out. Over here, I recommend you try the `InkWell` widget to see the differences between them:

For the `InkWell` widget, it has a nice ink-speading effect when tapping on it:

image

Please check the full document [here](https://api.flutter.dev/flutter/material/InkWell-class.html) if you want.

So now, let's give it a try. 😄

File: lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+              child: Text(
+                style: uiModel.isSelected
+                    ? Theme.of(context)
+                        .textTheme
+                        .displaySmall
+                        ?.copyWith(color: Colors.white)
+                    : Theme.of(context)
+                        .textTheme
+                        .headlineSmall
+                        ?.copyWith(color: Colors.white70),
```
Comment (nmint8m) [inline lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart:67]:
```suggestion
                    ? context.textTheme.displaySmall
                        ?.copyWith(color: Colors.white)
                    : context.textTheme.headlineSmall
                        ?.copyWith(color: Colors.white70),
```

File: lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    );
+  }
+
+  Widget _listItem(MultipleChoiceOptionUIModel uiModel) => SizedBox(
+        width: context.screenSize.width - 160.0,
+        height: Dimensions.multipleChoiceItemHeight,
+        child: Row(
+          children: [
+            Expanded(
+              child: Text(
```
Comment (nmint8m) [inline lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart:58]: Please extract the text widget into a class variable:

```suggestion
            Expanded(child: _titleText(context)),
```

```dart
    Widget  _titleText(BuildeContext context) => Text(...
```

File: lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+                      child: const FittedBox(
+                        child: Icon(
+                          Icons.check,
+                          size: 25.0,
+                          color: Colors.black,
+                        ),
+                      ),
+                    )
+                  : Container(),
+            ),
```
Comment (nmint8m) [inline lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart:96]: Please also extract the `_selectionBox` widget.

File: lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                      ),
+                      child: const FittedBox(
+                        child: Icon(
+                          Icons.check,
+                          size: 25.0,
+                          color: Colors.black,
+                        ),
+                      ),
+                    )
+                  : Container(),
```
Comment (nmint8m) [inline lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart:95]: Please use `const SizedBox.shrink()`

File: lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+          ],
+        ),
+      );
+
+  Widget _separatorBuilder({
+    required int count,
+    required int index,
+  }) {
+    if (index == count - 1) {
+      return Container();
```
Comment (nmint8m) [inline lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart:106]: Please use `const SizedBox.shrink()`

File: lib/ui/survey_question/survey_answer_views/text_field_form_view.dart [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+          index++;
+          answers[uiModel.id] = '';
+          return Container(
+            padding: const EdgeInsets.symmetric(
+                vertical: Dimensions.paddingSmallest),
+            child: TextFormField(
+              keyboardType: TextInputType.text,
+              decoration: _inputDecoration(
+                hintText: uiModel.title,
+              ),
```
Comment (nmint8m) [inline lib/ui/survey_question/survey_answer_views/text_field_form_view.dart:76]: As there's one param, please remove the `,`:

```suggestion
              decoration: _inputDecoration(hintText: uiModel.title),
```

**Review by nmint8m — COMMENTED** [review]
@ducbm051291 Next time, please help me to split the PR based on the story 😄

File: lib/ui/survey_question/survey_answer_views/emoticon_view.dart (hunk 1 above)
Comment (ducbm051291) [reply lib/ui/survey_question/survey_answer_views/emoticon_view.dart:27]: Amazing, let me give it a try

**Review by thiennguyen0196 — APPROVED** [review]
lgtm

**Review by nmint8m — APPROVED** [review]
Look good to me.

**Pull request merged**